ui/src/app/utils/authentication.tsx

```diff
@@ -92,3 +91,37 @@ export function useAuthentication() {
 
   return {authLoaded, isSignedIn};
 }
+
+// The delay before continuing to avoid errors due to delays in applying the new scope grant
+const BILLING_SCOPE_DELAY_MS = 2000;
+
+const getAuthInstance = () => {
+  return gapi.auth2.getAuthInstance();
+};
+
+export const hasBillingScope = () => {
+  return getAuthInstance().currentUser.get().hasGrantedScopes('https://www.googleapis.com/auth/cloud-billing');
+};
+
+
+export const delay = ms => {
+  return new Promise(resolve => setTimeout(resolve, ms));
+};
+
+/*
+ * Request Google Cloud Billing scope if necessary.
+ *
+ * NOTE: Requesting additional scopes may invoke a browser pop-up which the browser might block.
+ * If you use ensureBillingScope during page load and the pop-up is blocked, a rejected promise will
+ * be returned. In this case, you'll need to provide something for the user to deliberately click on
+ * and retry ensureBillingScope in reaction to the click.
+ */
+export const ensureBillingScope = async() => {
+  if (!hasBillingScope()) {
+    const options = new gapi.auth2.SigninOptionsBuilder();
+    options.setScope('https://www.googleapis.com/auth/cloud-billing');
+    await getAuthInstance().currentUser.get().grant(options);
+    // Wait 250ms before continuing to avoid errors due to delays in applying the new scope grant
```

Comment:
This is unfortunate. Do you observe this happening in practice? I'm curious whether simply `await delay(0)` would also do it.

Reply:
Seems 0 is not enough. I am not sure what is the best number, copied from:
https://github.com/DataBiosphere/terra-ui/pull/1318#discussion_r269305083 (they also increase to 250 from 100ms)